rdb/Query.php

```diff
@@ -56,8 +57,18 @@ public function encodeServerRequest()
         return array($this->getTermType(), $args, (object)$optargs);
     }
 
-    public function run(Connection $connection, $options = null)
+    public function run(Connection $connection = null, $options = array())
     {
+        if (!($connection instanceof Connection)) {
+            $options = is_array($connection) ? $connection : $options;
+
+            if (!isset($this->connection)) {
```

Comment:
I don't quite understand how this works if no `$connection` is set when calling `run`. This seems to fall back to the `$this->connection` member variable of `Query`, but as far as I can tell `$this->connection` is never set anywhere.
(Is this just code that is remaining from when you wanted to make the connection inherited through queries?)